test_ai/main.go

```diff
@@ -0,0 +1,202 @@
+package main
+
+import (
+	"fmt"
+	sdz "github.com/mzimmerman/sdzpinochle"
+	ai "github.com/mzimmerman/sdzpinochle/ai"
+	"sort"
+)
+
+const (
+	winningScore   int  = 120
+	numberOfTricks int  = 12
+	debugLog       bool = false
+)
+
+type NamedBid struct {
+	Name string
+	Bid  ai.BiddingStrategy
+}
+
+type NamedPlay struct {
+	Name string
+	Play ai.PlayingStrategy
+}
+
+func main() {
+	sdz.Init()
+	bidding_strategies := []NamedBid{
+		NamedBid{"NeverBid", ai.NeverBid},
+		NamedBid{"MostMeld", ai.ChooseSuitWithMostMeld},
+	}
+	playing_strategies := []NamedPlay{
+		NamedPlay{"PlayHighest", ai.PlayHighest},
+		NamedPlay{"PlayLowest", ai.PlayLowest},
+		NamedPlay{"PlayRandom", ai.PlayRandom},
+	}
+	players := make([]ai.Player, 0)
+	for _, b := range bidding_strategies {
+		for _, p := range playing_strategies {
+			players = append(players, ai.Player{fmt.Sprintf("%v:%v", b.Name, p.Name), b.Bid, p.Play})
+		}
+	}
+	player1 := players[0]
+	for _, player2 := range players[1:] {
+		win0 := 0
+		win1 := 0
+		fmt.Printf("%v vs %v\n", player1.Name, player2.Name)
+		for x := 0; x < 10000; x++ {
+			winningPartnership, match := playMatch(player1, player2)
+			if winningPartnership == 0 {
+				win0++
+			} else {
+				win1++
+			}
+			if debugLog {
+				fmt.Printf("Partnership %v won! %v\n", winningPartnership, match)
+			}
+		}
+		fmt.Printf("%v %v wins - %v %v wins\n", player1.Name, win0, player2.Name, win1)
+		if win1 > win0 {
+			// Winner stays
+			player1 = player2
+		}
+	}
+	fmt.Printf("%v is the champ!\n", player1.Name)
+
+}
+
+func playMatch(player1, player2 ai.Player) (int, *ai.Match) {
+	var players [4]ai.Player
+	for x := 0; x < 4; x++ {
+		if x%2 == 0 {
+			players[x] = player1
+		} else {
+			players[x] = player2
+		}
+	}
+	match := ai.Match{
+		Partnerships: new([2]ai.Partnership),
+		Players:      players,
+	}
+	winner := -1
+	for x := 1; winner == -1; x++ {
+		bidder := playDeal(&match, x%4)
+		pOne := match.Partnerships[0]
+		pTwo := match.Partnerships[1]
+		if pOne.MatchScore > winningScore && pTwo.MatchScore > winningScore {
```

Comment:
I also see you've taken Uncle John's approach -- must be *over* 120, but only when it's your opponent that has 120 :)